src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -178,122 +177,157 @@ def apply(
 
         if isinstance(node, ast.Subscript):
             if isinstance(node.slice, ast.Index):
-                slice_node = cls.single_index_slice(node.slice.value)
+                slice_node = cls.slice_from_value(node.slice.value)
             else:
                 slice_node = node.slice
         elif isinstance(node, ast.Slice):
             slice_node = node
         else:
-            slice_node = cls.single_index_slice(node)
+            slice_node = cls.slice_from_value(node)
 
-        if slice_node.lower is not None:
-            start = parser.visit(slice_node.lower)
-        else:
-            start = gt_ir.AxisBound(level=gt_ir.LevelMarker.START, offset=0, extend=True)
+        if slice_node.lower is None:
+            slice_node.lower = ast.Constant(value=None)
 
-        if slice_node.upper is not None:
-            end = parser.visit(slice_node.upper)
-        else:
-            end = gt_ir.AxisBound(level=gt_ir.LevelMarker.END, offset=0, extend=True)
+        if (
+            isinstance(slice_node.lower, ast.Constant)
+            and slice_node.lower.value is None
+            and axis_name == gt_ir.Domain.LatLonGrid().sequential_axis.name
+        ):
+            raise parser.interval_error
+
+        lower = parser.visit(slice_node.lower)
+
+        if slice_node.upper is None:
+            slice_node.upper = ast.Constant(value=None)
+        upper = parser.visit(slice_node.upper)
+
+        start = parser._make_axis_bound(lower, gt_ir.LevelMarker.START)
+        end = parser._make_axis_bound(upper, gt_ir.LevelMarker.END)
 
         return gt_ir.AxisInterval(start=start, end=end, loc=loc)
 
     def __init__(
         self,
         axis_name: str,
-        context: Optional[dict] = None,
         loc: Optional[gt_ir.Location] = None,
     ):
         self.axis_name = axis_name
-        self.context = context or dict()
         self.loc = loc
 
-        # initialize possible exceptions
-        self.interval_error = ValueError(
-            f"Invalid 'interval' specification at line {loc.line} (column {loc.column})"
-        )
+        error_msg = "Invalid 'interval' specification"
+
+        if self.loc is not None:
+            error_msg = f"{error_msg} (line: {loc.line}, column: {loc.column})"
+
+        self.interval_error = GTScriptSyntaxError(error_msg)
 
     @staticmethod
-    def single_index_slice(node: ast.Expr):
+    def slice_from_value(node: ast.Expr) -> ast.Slice:
```

Comment:
I see your point, @tehrengruber, but I'm not so sure about your proposal. The region specification here is still 2D, even if one of the axes only contains one element, and although your proposal is slightly more correct from a mathematical point of view, it's also more verbose and therefore harder to read. Personally, I'd accept the current succinct specification for these cases, since changing the implementation here also involves changing the syntax in the GDP, but I might be wrong.

Reply:
I pondered with the syntax a little bit more and I'm ok with it now. My reasoning here is/was two-fold. First the syntax adds additional complexity to the backend, which I would like to avoid until it is overhauled at some point. I still believe this is a valid point, but a matter of taste. Secondly I understood the syntax (symbolically) means `region[...]` returns the slices of the domain to run on, in which case it should not change its dimension. However one can also see it as returning the indices of the domain to run on, where the dimension doesn't matter. I'm not in favor of this perspective, but it's valid one and the verbosity is also a good point.